lib/array.rb

```diff
@@ -0,0 +1,32 @@
+class Array
+
+  def mick_inject(initial = nil, &block)
+    return nil if count == 0
+    if all?{ |a| !a }
+      accumulator = initial
+      each do |a|
+        accumulator = block.call(accumulator, a)
+      end
+      return accumulator
+    end
+
+    if !initial
+      initial = self[0]
+      shift
+      mick_inject(initial, &block)
+    else
+      accumulator = initial
+      each do |a|
+        accumulator = block.call(accumulator, a)
+      end
+      accumulator
+    end
+  end
+
+end
+
+array = [nil,nil,nil,nil]
+
+p array.count
+p array.inject{|acc, x| acc = !x}
```

Comment:
Lint/UnusedBlockArgument: Unused block argument - `acc`. If it's necessary, use `_` or `_acc` as an argument name to indicate that it won't be used.<br>Rails/Output: Do not write to stdout. Use Rails' logger if you want to log.